lighting-estimation-explainer.md

```diff
@@ -0,0 +1,91 @@
+
+This "SH probe" format is used by most modern rendering engines, including Unity, Unreal, and Threejs.
+
+## Shadows
+
+When an HDR Cube Map texture is available, shadows only have to consider occlusion of other rendered objects in the scene.
+
+When a HDR Cube Map texture is not available, or the typical soft shadow effects of image based lighting are too costly to implement, the XRLightProbe.primaryLightDirection and XRLightProbe.primaryLightIntensity can be used to render shadows cast by the most prominent light source.
+
+## Security Implications
+
+### XRLightProbe
+
+Only XRLightProbe.indirectIrradiance is guaranteed to be available either due to user privacy settings or the capabilities of the platform.
+
+XRLightProbe returns sufficient information to render objects that appear to fit into their environment, with highly diffuse surfaces or high frequency normal maps which would result in a wide NDF (normal distribution function).  Highly polished objects may be represented with a non-physically based illusion of glossiness with a specular highlight effect sensitive only to the primary light direction.  Reflections will be unable to reproduce detailed images of the environment without an XRReflectionProbe.
+
+The lighting estimation returned by the WebXR API explicitly describes the real world environment in proximity to the user.  By default, only low spatial frequency and low temporal frequency information should be returned by the WebXR API.  Even when a platform can directly produce higher spatial and temporal frequency information, the browser must apply a low pass filter with an aim to mitigate the risk of untrusted content identifying the geolocation of the user or of profiling their environment.
+
+Combined with other factors, such as the user's IP address, even the low frequency information returned with XRLightProbe increases the fingerprinting risk.  The XRLightProbe should only be accessible during an active WebXR session.
+
+### XRReflectionProbe
+
+XRReflectionProbe should only be accessible with a permissions prompt equivalent to requesting access to the camera and microphone.  XRReflectionProbe enables efficient and simple to implement image based lighting.  PBR shaders can index the mip map chain of the environment cube to reduce the memory bandwidth required while integrating multiple samples to match wider NDF's.
+
+## Appendix A: Proposed partial IDL
+This is a partial IDL and is considered additive to the core IDL found in the main [explainer](explainer.md).
+
+```webidl
+partial interface XRFrame {
+  Promise<XRLightProbe> getGlobalLightEstimate();
+  Promise<XRReflectionProbe> getGlobalReflectionProbe();
```

Comment:
Do we want this API to be promise-based? If the lighting is estimated for a specific frame and we're returning promises, then the application would not be able to get the result of `getGlobalLightEstimate()` / `getGlobalReflectionProbe()` during request animation frame callback relevant for the frame for which the lighting was estimated. It would be able to act on the data (at the earliest) in a subsequent frame's rAF, and by that time, the estimate might be outdated.

Reply:
One alternative would be to make this API subscription-based (similarly to what's described in existing [hit test explainer](https://github.com/immersive-web/webxr/blob/master/hit-testing-explainer.md)).